src/compiler/checker.ts

```diff
@@ -21746,6 +21749,37 @@ namespace ts {
             }
         }
 
+        function addIntraExpressionInferenceSite(context: InferenceContext, node: Expression | MethodDeclaration, type: Type) {
+            (context.intraExpressionInferenceSites ??= []).push({ node, type });

```

Comment:
If there are multiple nested active inference contexts (eg, `foo(..., bar(..., baz(..., {...})))` might we not want to add the expression to all inference contexts? This way if a type parameter flows from `foo` into `bar` and then `baz`, but doesn't flow back out on construction (eg, because it's conditional'd away), we could still pick up the inner inference site.

Reply:
No, context sensitive arguments to an inner call might cause an outer type parameter to become fixed, but _inferences_ from the inner argument expressions only occur through the return type of the inner argument expressions.